sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgInstructionFactoryTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿extern alias csharp;
+/*
+* SonarAnalyzer for .NET
+* Copyright (C) 2015-2018 SonarSource SA
+* mailto: contact AT sonarsource DOT com
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+using System;
+using csharp::SonarAnalyzer.ControlFlowGraph.CSharp;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarAnalyzer.ControlFlowGraph;
+
+namespace SonarAnalyzer.UnitTest.ControlFlowGraph
+{
+    [TestClass]
+    public class UcfgInstructionFactoryTests
+    {
+        [TestMethod]
+        public void CreateFrom_DecoratesAndReThrowsExceptions()
+        {
+            // Arrange
+            var mockSemanticModel = new Mock<SemanticModel>(MockBehavior.Strict); // strict -> throw on any usage
```

Comment:
This should push the code coverage high enough (although there are still uncovered lines in some fairly complex bits of the file that may or may not be required or working as expected).

The instruction factory and expression service aren't easy to test; they're internal so they can't be mocked and they don't have virtual methods, so it's difficult to inject behaviours (e.g. throwing an exception when testing).

_SyntaxNode_ can't be mocked either since it only has internal constructors.
Fortunately, _SemanticModel_ is public and abstract so it can be mocked.

I looked at creating an _IUcfgExpressionService_ interface, but that would have involved a lot more churn for very little value (also, the _UcfgExpressionService_ uses default arguments for a couple of methods, so I'd had to have added extra methods or changed all the callers).